src/Assets.php

```diff
@@ -182,8 +182,8 @@ public static function get_wc_block_data( $settings ) {
 					'privacy'  => self::format_page_resource( $page_ids['privacy'] ),
 					'terms'    => self::format_page_resource( $page_ids['terms'] ),
 				],
-				'checkoutAllowsGuest'           => 'yes' === get_option( 'woocommerce_enable_guest_checkout' ),
-				'checkoutAllowsSignup'          => 'yes' === get_option( 'woocommerce_enable_signup_and_login_from_checkout' ),
+				'checkoutAllowsGuest'           => filter_var( get_option( 'woocommerce_enable_guest_checkout' ), FILTER_VALIDATE_BOOLEAN ),
+				'checkoutAllowsSignup'          => filter_var( get_option( 'woocommerce_enable_signup_and_login_from_checkout' ), FILTER_VALIDATE_BOOLEAN ),
```

Comment:
`checkoutAllowsSignup` (store setting) is no longer needed by the block - could remove this, or use it for the default value of the block option.

Reply:
Now using the store setting for the default value of the block option. 5a3c1f99c16566dc913ee440e2131b068e0f2e13